packages/xo-web/src/common/utils.js

```diff
@@ -371,13 +370,20 @@ export const resolveResourceSets = resourceSets =>
 //   id: 42,
 // }, 32)
 //
-// newString === 'foo_COPY_foo_42_32'
+// newString === 'foo_COPY_{name}_42_32%'
 // ```
 export function buildTemplate(pattern, rules) {
-  const regExp = new RegExp(join(map(keys(rules), escapeRegExp), '|'), 'g')
+  const transforms = { '\\\\': '\\' }
+  const matches = ['\\\\']
+  forOwn(rules, (output, input) => {
+    transforms[input] = output
+    transforms[`\\${input}`] = input
+    matches.push(input, `\\${input}`)
+  })
+  const regExp = new RegExp(matches.map(escapeRegExp).join('|'), 'g')
```

Comment:
Here is an alternate implementation with more optimized regexp and not relying on a new `transform` map:

```js
const matches = Object.keys(rules).sort(compareLengthDesc).map(escapeRegExp).join('|')
const regExp = new RegExp(
  `\\\\(?:${matches})|${matches}`,
  'g'
)
return (...params) =>
  pattern.replace(regExp, match => {
    if (match[0] === '\\') {
      return match.slice(1)
    }
    const rule = rules[match]
    return typeof rule === 'function' ? rule(...params) : rule
  })
```

Reply:
Should i handle the case which it have a backslash followed by a template variable?

Reply:
Yes, it should be supported.

IMHO we should move this function in its own module and add some tests.